src/core/a-node.js

```diff
@@ -32,19 +33,13 @@ class ANode extends HTMLElement {
     this.mixinEls = [];
   }
 
-  onReadyStateChange () {
-    if (document.readyState === 'complete') {
-      this.doConnectedCallback();
-    }
-  }
-
   connectedCallback () {
     // Defer if DOM is not ready.
-    if (document.readyState !== 'complete') {
-      document.addEventListener('readystatechange', this.onReadyStateChange.bind(this));
+    if (!ready.isReady) {
+      document.addEventListener('aframeready', this.connectedCallback.bind(this));
       return;
     }
-    ANode.prototype.doConnectedCallback.call(this);
```

Comment:
Any reason why this call has changed?

Reply:
Yeah, the inheritance was a bit wonky. The `doConnectedCallback` function was essentially setup as a "virtual function" with subclasses overriding it and calling the base implementation (`super.doConnectedCallback()`), yet `connectedCallback` explicitly called the `ANode` implementation, forcing all subclasses to also override `connectedCallback` and replicating the same logic. From an OOP standpoint this didn't really make sense. Having one shared `connectedCallback` implementation is the cleanest solution IMO.